extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -687,14 +696,40 @@ export class Bind {
    * Begin observing mutations to element. Presently, all supported elements
    * that can add/remove bindings add new elements to their parent, so parent
    * node should be observed for mutations.
+   * @param {!Element} element
    * @private
    */
   observeElementForMutations_(element) {
-    // TODO(kmh287): What if parent is the body tag?
-    // TODO(kmh287): Generify logic for node observation strategy
-    // when bind supprots more dynamic nodes.
-    const elementToObserve = element.parentElement;
-    this.mutationObserver_.observe(elementToObserve, {childList: true});
+    const tagName = element.tagName;
+    let elementToObserve;
+    if (tagName === 'TEMPLATE') {
+      // Templates add templated elements as siblings of the template tag
+      // so the parent must be observed.
+      // TODO(kmh287): What if parent is the body tag?
+      elementToObserve = element.parentElement;
+    } else if (tagName === 'AMP-LIVE-LIST') {
+      // All elements in AMP-LIVE-LIST are children of a <div> with an
+      // `items` attribute.
```

Comment:
`dom.js#childElementByAttr`. Found that by looking at the implementation of `amp-live-list.js`.

I wonder if we should delegate this to the dynamic element itself, similar to `mutatedAttributesCallback`.